### ISSUE TITLE
Delay when unstacking windows

### ISSUE THREAD

**Issue by diocletiann** [koekeishiya/yabai#2188]
This is my "unstack" command:
```
yabai -m window --insert [SIDE] --toggle float --toggle float
``` 
It's been working great for a long time. Recently I'm noticing a ~1s delay under certain conditions, for example when there are multiple windows that get resized as a result and they are not full display width. I tried using two separate commands for insert and the two toggles and it didn't change the behavior.

**Comment by koekeishiya** [comment]
Resize is just a slow operation for some windows depending on which content they are drawing, and how the window frame changes. See for example this comment https://github.com/koekeishiya/yabai/issues/2137#issuecomment-1980643218

**Comment by diocletiann** [comment]
Fair enough. Only bringing this up because it happens in a space with only Alacritty windows which hadn't caused issues before.

**Comment by koekeishiya** [comment]
Nothing code-wise has changed for these operations that I can think of, so I'm not sure what exactly would be the cause if it is now happening a lot more frequently than before. 

You can build yabai from source with profiling information if you want to try and identify patterns, and where time is spent. 
Basically, when compiling the code run `make install CLI_FLAGS=-DPROFILE=2` instead of just `make install`.
You can do `brew edit yabai` and change the `make install` line in the formula if you brefer to use homebrew.

Sample outout when built with profiling samples, like with `window_animation_duration` enabled:

```
EVENT_HANDLER_DAEMON_MESSAGE: window --insert east --toggle float --toggle float
Total time: 129.8613ms (timer freq 24000000)
    EVENT_HANDLER_DAEMON_MESSAGE[1]: 761 (0.02%, 100.00% w/children)
    parse_window_selector[1]: 135 (0.00%)
    handle_domain_window[1]: 362 (0.01%, 99.97% w/children)
    window_manager_animate_window_list_async___prep_proxies[2]: 153151 (4.91%)
    window_manager_animate_window_list_async___wait_for_threads[2]: 1549814 (49.73%)
    window_manager_animate_window_list_async___swap_proxy_in[2]: 14875 (0.48%)
    window_manager_animate_window_list_async___notify_jb[2]: 19383 (0.62%)
    window_manager_animate_window_list_async___set_frame[2]: 535353 (17.18%)
    window_manager_animate_window_list[2]: 196952 (6.32%, 79.24% w/children)
    window_manager_focused_application[2]: 888 (0.03%)
    window_manager_focused_window[2]: 54307 (1.74%, 1.77% w/children)
    window_manager_set_window_insertion[1]: 319597 (10.25%)
    window_manager_make_window_floating[2]: 271070 (8.70%, 88.02% w/children)
```
 and without animations:

```
EVENT_HANDLER_DAEMON_MESSAGE: window --insert east --toggle float --toggle float
Total time: 62.8209ms (timer freq 24000000)
    EVENT_HANDLER_DAEMON_MESSAGE[1]: 1177 (0.08%, 100.00% w/children)
    parse_window_selector[1]: 77 (0.01%)
    handle_domain_window[1]: 321 (0.02%, 99.92% w/children)
    window_manager_animate_window_list[2]: 672735 (44.62%)
    window_manager_focused_application[2]: 859 (0.06%)
    window_manager_focused_window[2]: 27365 (1.82%, 1.87% w/children)
    window_manager_set_window_insertion[1]: 508561 (33.73%)
    window_manager_make_window_floating[2]: 296543 (19.67%, 64.54% w/children)
```

In the latter case, 44.62% of time spent (out of 62ms), was spent calling the AX API to reposition and resize windows, and this is time we cannot improve. The other operations that happen here can likely be improved somewhat. I haven't yet cataloged all this and identified where optimizations can take place yet.

Obviously these numbers can vary between runs for the exact same operations for various reasons; systems today are complex.

The long term goal is of course to try and look at these things in detail to identify commonly used areas that are feasible to make more performant.

**Comment by diocletiann** [comment]
Thanks for the details and for all the work on Yabai.

**Comment by diocletiann** [comment]
Sorry, hit the wrong button and closed the ticket. It seems macOS "freezes" during this pause, if I have a terminal window open even the window that isn't getting resized doesn't print anything until the pause ends. It also happens when using during `insert + warp`. I'll try to build with profiling enabled and repro.

**Comment by koekeishiya** [comment]
Something with the `--insert` command *sometimes* causes the system API `SLSCopyWindowsWithOptionsAndTags` to take an insane amount of time, but only if the warp command happen rapidly and sequentially. Invoking `yabai -m window --insert .. && yabai -m window --warp ..` does not cause the system API to take forever to respond. Weird.

**Comment by diocletiann** [comment]
Hmm, I am sending `insert` followed by a `warp` in rapid succession to the yabai socket...

**Comment by koekeishiya** [comment]
I think the `--insert .. --warp ..` chain should be resolved on master. Not sure what is going on with the `--insert .. --toggle float` part. If you could get some profiling information for that sequence, it could prove helpful.

**Comment by diocletiann** [comment]
Thank you. I tried to reproduce both `--insert .. --warp ..` and `--insert .. --toggle float` and I'm not seeing any delays. I'll rebuild with profiling enabled later today.

**Comment by koekeishiya** [comment]
Released 7.0.4. Hopefully this is resolved. Feel free to create a new issue if you happen to notice such issues in the future.

**Comment by diocletiann** [comment]
> Released 7.0.4. Hopefully this is resolved. Feel free to create a new issue if you happen to notice such issues in the future.

I'm not able to repro this issue anymore, thank you!
